cpu/cortexm_common/thread_arch.c

```diff
@@ -234,6 +238,17 @@ char *thread_arch_stack_init(thread_task_func_t task_func,
     return (char*) stk;
 }
 
+/* This function gets the number of bytes used on the ISR stack */
+int thread_arch_isr_stack_usage(void)
+{
+    int count = 0;
+    uint32_t* sp = (uint32_t *) _estack;
+    while(sp != (uint32_t *) _sstack)
+        if (*sp == STACK_CANARY_WORD)
+            count++;
+    return ((int)(512 - count));
```

Comment:
makes sense: @MohmadAyman in any case: please document any extra meaning of return values in the function prototype!

@kaspar030: I think we made one mistake: the return value must be
```c
return ISR_STACK_SIZE - (ptr - &_sstack);
```
as we want to return the stack space that is used, right?